dokka-integration-tests/gradle/src/testExampleProjects/kotlin/ExampleProjectsTest.kt

```diff
@@ -0,0 +1,344 @@
+/*
+ * Copyright 2014-2024 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+package org.jetbrains.dokka.it.gradle.examples
+
+import io.kotest.assertions.asClue
+import io.kotest.assertions.withClue
+import io.kotest.matchers.nulls.shouldNotBeNull
+import io.kotest.matchers.paths.shouldBeADirectory
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldContain
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome.UP_TO_DATE
+import org.jetbrains.dokka.gradle.utils.*
+import org.jetbrains.dokka.it.gradle.loadConfigurationCacheReportData
+import org.jetbrains.dokka.it.systemProperty
+import org.junit.jupiter.api.Assumptions.assumeTrue
+import org.junit.jupiter.api.Named.named
+import org.junit.jupiter.api.extension.ExtensionContext
+import org.junit.jupiter.api.io.TempDir
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.Arguments.arguments
+import org.junit.jupiter.params.provider.ArgumentsProvider
+import org.junit.jupiter.params.provider.ArgumentsSource
+import java.nio.file.Path
+import java.util.stream.Stream
+import kotlin.io.path.*
+import kotlin.streams.asStream
+
+class ExampleProjectsTest {
+
+    class TestCaseProvider : ArgumentsProvider {
+
+        override fun provideArguments(context: ExtensionContext): Stream<Arguments> {
+            val exampleProjectDirs = exampleGradleProjectsDir
```

Comment:
It might make sense to check that `exampleProjectDirs` is not empty or these tests might silently become noop in the future

Reply:
If the example project directory is empty then the test will fail.

```
* What went wrong:
Directory '/private/var/folders/_3/ps18c9f56bx2j4fyxt0z7sn40000gp/T/9203619320687405921/basic-gradle-example' does not contain a Gradle build.
```